bluetoe/link_layer/include/bluetoe/peripheral_latency.hpp

```diff
@@ -0,0 +1,318 @@
+#ifndef BLUETOE_LINK_LAYER_PERIPHERAL_LATENCY_HPP
+#define BLUETOE_LINK_LAYER_PERIPHERAL_LATENCY_HPP
+
+#include <bluetoe/ll_meta_types.hpp>
+
+/**
+ * @file bluetoe/peripheral_latency.hpp
+ *
+ * This file provides several options, how Bluetoe will implement peripheral latency. Peripheral latency
+ * allows a link layer peripheral to miss some connection events (do not respond to the central) to
+ * conserve power.
+ *
+ * Each option is a compromise between latency and power consumption.
+ *
+ * @sa bluetoe::link_layer::peripheral_latency_ignored
+ * @sa ...
+ */
+namespace bluetoe {
+namespace link_layer {
+
+    namespace details {
+        struct peripheral_latency_meta_type {};
+    }
+
+    /**
+     * @brief detailed options for the peripheral latency behavior
+     *
+     * Every option defines a set of events / circumstances under which
+     * the link layer should listen to the very next connection event.
+     * (beside the minumum events, the link layer has to listen to due
+     * to the periperhal latency value of the connection).
+     *
+     * @note in combination, there are some options that make no sense.
+     *       For example, using listen_always with some of the other
+     *       listen_* options is pointless. Also combining send_always
+     *       with send_pending_data doesn not make sense.
+     */
+    enum class peripheral_latency
+    {
+        /**
+         * @brief listen at all connection events.
+         *
+         * This does not imply, that the link layer will also respond at every
+         * connection event.
+         */
+        listen_always,
+
+        /**
+         * @brief listen, if the link layer contains unacknowlaged data.
+         *
+         * If a PDU was send out, but no acknowlagement was jet received,
+         * the link layer will listen at the next connection event, if
+         * this options is set.
+         *
+         * This might be interesting, if the sending queue is very small because
+         * Bluetoe has to keep a copy of a transmitted PDU, as long as that PDU
+         * is not acknowlaged.
+         */
+        listen_if_unacknowlaged_data,
+
+        /**
+         * @brief listen to the next connection event, if the last received PDU was
+         *        not empty.
+         */
+        listen_if_last_received_not_empty,
+
+        /**
+         * @brief listen to the next connection event, if the last transmitted PDU was
+         *        not empty.
+         */
+        listen_if_last_transmitted_not_empty,
```

Comment:
Same as above

Reply:
Is it unclear, or what?

Reply:
I think that @xnlcasad means that if we remove the listen_if_last_received_not_empty then we should also remove the listen_if_last_transmitted_not_empty.

Also,listen_if_last_transmitted_not_empty would be a mechanism similar to the N one but for transmit, which I don't think we would need in any case because we already have listen_if_pending_data which will take care of the pure stack part (i.e., making sure that pending data gets transmitted ASAP).